x-pack/server/lib/create_router/call_with_request_factory/call_with_request_factory.js

```diff
@@ -6,13 +6,12 @@
 
 import { once } from 'lodash';
 
-const callWithRequest = once(server => {
-  const cluster = server.plugins.elasticsearch.getCluster('data');
-  return cluster.callWithRequest;
-});
-
 export const callWithRequestFactory = (server, request) => {
   return (...args) => {
-    return callWithRequest(server)(request, ...args);
+    const callWithRequest = server => {
+      const cluster = server.plugins.elasticsearch.getCluster('data');
+      return cluster.callWithRequest;
+    };
+    return once(callWithRequest)(server)(request, ...args);
```

Comment:
I'm sure we've had discussion about this before and I just missed it, but what's the benefit of using `once` here? If it's not needed, then this code could be simplified a bit. I tested this locally and it seemed OK:

```js
export const callWithRequestFactory = (server, request) => {
  const { callWithRequest } = server.plugins.elasticsearch.getCluster('data');
  return (...args) => {
    return callWithRequest(request, ...args);
  };
};
```

Reply:
`once` was used in the original code to safeguard against multiple factory calls. now that the factory code for call with request and license checker is shared, you're right that it is no longer needed. I pushed an update to remove `once` from both factories